src/libasr/asr_utils.cpp

```diff
@@ -713,7 +714,69 @@ ASR::asr_t* symbol_resolve_external_generic_procedure_without_eval(
     }
 }
 
-    } // namespace ASRUtils
+ASR::asr_t* make_ImplicitCast_t_value(Allocator &al, const Location &a_loc, ASR::expr_t* a_arg, ASR::cast_kindType a_kind, ASR::ttype_t* a_type) {
+
+    ASR::expr_t* value = nullptr;
+
+    if (ASRUtils::expr_value(a_arg) != nullptr && value == nullptr){
```

Comment:
`value` is always `nullptr` here:
```suggestion
    if (ASRUtils::expr_value(a_arg)) {
```